### PR TITLE
Make PostMappingBuildEvent extends IndexEvent

### PR DESCRIPTION
**Pull request by deguif**
This would allow to easily check if event needs to be handled for a specific index, by using for example:

```php
if ('my_index' === $event->getIndex()) {
    // do stuff
}
```

### CODE REVIEW

File: src/Event/PostMappingBuildEvent.php [hunk 1]
```diff
@@ -12,9 +12,8 @@
 namespace FOS\ElasticaBundle\Event;
 
 use FOS\ElasticaBundle\Configuration\IndexConfigInterface;
-use Symfony\Contracts\EventDispatcher\Event;
 
-final class PostMappingBuildEvent extends Event
+final class PostMappingBuildEvent extends IndexEvent
```
Comment (deguif) [inline src/Event/PostMappingBuildEvent.php:16]: Maybe we should also rename it to PostIndexMappingBuildEvent?

ping @darthf1

File: src/Event/PostMappingBuildEvent.php [hunk 2]
```diff
@@ -30,6 +29,8 @@ public function __construct(IndexConfigInterface $indexConfig, array $mapping)
     {
         $this->indexConfig = $indexConfig;
         $this->mapping = $mapping;
+
+        parent::__construct($indexConfig->getName());
```
Comment (deguif) [inline src/Event/PostMappingBuildEvent.php:33]: I'm always confused about which name to use, should it be `$indexConfig->getElasticsearchName()`?
ping @XWB

File: src/Event/PostMappingBuildEvent.php (hunk 1 above)
Comment (darthf1) [reply src/Event/PostMappingBuildEvent.php:16]: I have no opinion on this. It makes sense if you extend the `IndexEvent`

Comment (XWB) [reply src/Event/PostMappingBuildEvent.php:16]: Yeah, `PostIndexMappingBuildEvent` would be better.

File: src/Event/PostMappingBuildEvent.php (hunk 2 above)
Comment (XWB) [reply src/Event/PostMappingBuildEvent.php:33]: There is no `getElasticsearchName` method, do you mean renaming `getName()` to `getElasticsearchName()`?

Comment (deguif) [reply src/Event/PostMappingBuildEvent.php:33]: I mean this method: https://github.com/FriendsOfSymfony/FOSElasticaBundle/blob/master/src/Configuration/IndexConfigInterface.php#L21 vs this one https://github.com/FriendsOfSymfony/FOSElasticaBundle/blob/master/src/Configuration/IndexConfigInterface.php#L25

There is an abstraction of the name of the real index name that can be achieved with the bundle configuration, when using for example:
```yaml
fos_elastica:
    indexes:
        app:
            index_name: app_prod
```

`getName()` would return in this case `app` where `getElasticsearchName()` would return `app_prod`

Comment (XWB) [reply src/Event/PostMappingBuildEvent.php:33]: Good question. I have to admin I'm confused too. Perhaps we should stick with `getName()` as long as things work nicely.

Comment (deguif) [reply src/Event/PostMappingBuildEvent.php:33]: Yes, seems fair ;)

**Comment by deguif** [comment]
Oups it was merged too quickly, I had to rebase before to take into account other PR. I'm creating a new one.

**Pull request merged**